data_steward/cdr_cleaner/cleaning_rules/negative_ages.py

```diff
@@ -27,110 +32,261 @@
-        query_na[cdr_consts.DISPOSITION] = bq_consts.WRITE_TRUNCATE
-        query_na[cdr_consts.DESTINATION_DATASET] = dataset_id
-        query_ma[cdr_consts.QUERY] = MAX_AGE_QUERY.format(
-            project_id=project_id,
-            dataset_id=dataset_id,
-            table=table,
-            person_table=person_table,
-            table_date=date_fields[table],
-            MAX_AGE=MAX_AGE)
-        query_ma[cdr_consts.DESTINATION_TABLE] = table
-        query_ma[cdr_consts.DISPOSITION] = bq_consts.WRITE_TRUNCATE
-        query_ma[cdr_consts.DESTINATION_DATASET] = dataset_id
-        queries.extend([query_na, query_ma])
-
-    # query for death before birthdate
-    death = common.DEATH
-    query = dict()
-    person_table = person
-    query[cdr_consts.QUERY] = NEGATIVE_AGE_DEATH_QUERY.format(
-        project_id=project_id,
-        dataset_id=dataset_id,
-        table=death,
-        person_table=person_table)
-    query[cdr_consts.DESTINATION_TABLE] = death
-    query[cdr_consts.DISPOSITION] = bq_consts.WRITE_TRUNCATE
-    query[cdr_consts.DESTINATION_DATASET] = dataset_id
-    queries.append(query)
-    return queries
+NEGATIVE_AGE_DEATH_QUERY = JINJA_ENV.from_string("""
+SELECT
+  *
+FROM
+  `{{project_id}}.{{dataset_id}}.{{table}}`
+WHERE
+  person_id NOT IN (
+  SELECT
+    d.person_id
+  FROM
+    `{{project_id}}.{{dataset_id}}.{{table}}` d
+  JOIN
+    `{{project_id}}.{{dataset_id}}.{{person_table}}` p
+  ON
+    d.person_id = p.person_id
+  WHERE
+    d.death_date < DATE(p.birth_datetime))
+""")
+
+
+class NegativeAges(BaseCleaningRule):
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id=None):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Returns queries to remove table records which are prior to the persons birth date or 150 years past the '
+            'birth date from a dataset.')
+
+        super().__init__(issue_numbers=JIRA_ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[cdr_consts.COMBINED],
+                         affected_tables=affected_tables,
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_query_specs(self, *args, **keyword_args) -> query_spec_list:
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+
+        sandbox_queries = []
+        queries = []
```

Comment:
I'm not sure it helps to have a separate var for sandbox queries. I think they are distinguished well enough in the loop by the variable `sandbox_query` which can just be appended to `queries`. This could prevent the return type bug (see comment in unit test).